scripts/test_env.py

```diff
@@ -0,0 +1,191 @@
+        setup(args)
+    elif args.command == "cleanup":
+        cleanup(args)
+
+def setup(args):
+    logging.info("Starting iroha network with %d peers", args.n_peers)
+    setup_dir = pathlib.Path(args.out_dir)
+    try:
+        os.mkdir(setup_dir)
+    except FileExistsError:
+        logging.error("Test directory `%s` already exists", setup_dir)
+        sys.exit(1)
+    try:
+        shutil.copy2("./target/debug/iroha_client_cli", setup_dir)
+    except FileNotFoundError:
+        logging.error("Please build `iroha_client_cli` by running `cargo build --bin iroha_client_cli`")
+        sys.exit(1)
+    with open(os.path.join(setup_dir, "metadata.json"), "w") as f:
+        f.write('{"comment":{"String": "Hello Meta!"}}')
+    shutil.copy2("./configs/client/config.json", setup_dir)
+    try:
+        shutil.copy2("./target/debug/kagami", setup_dir)
+    except FileNotFoundError:
+        logging.error("Please build `kagami` by running `cargo build --bin kagami`")
+        sys.exit(1)
+
+    Network(setup_dir, args.host_addr, args.n_peers).run()
+
+def cleanup(args):
+    setup_dir = args.out_dir
+    logging.info("Killing peer processes...")
+    subprocess.run(["pkill", "-9", "iroha"])
+    logging.info("Cleaning up test directory `%s`", setup_dir)
+    shutil.rmtree(setup_dir)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description=__doc__)
+
+    parser.add_argument("command", choices=["setup", "cleanup"],
+                        help="Available actions. The `setup` command will create and run \
+                        a new test environment with specified number of peers in a \
+                        directory of choice. This is meant to be run from `iroha` root dir. \
+                        The `cleanup` command will kill all peer processes \
+                        that were started by the `setup` command and remove the test directory")
+    parser.add_argument("n_peers", nargs="?", default=4, type=pos_int,
+                        help="Number of peers to bootstrap. \
+                        Defaults to 4. If setup was run with a custom number of peers, \
+                        the same number doesn't need to be provided to cleanup as \
+                        it kills all processes named `iroha`, so proper caution is advised")
+
+    parser.add_argument("--out_dir", "-o", default="./test", type=pathlib.Path,
+                        help="Directory to store config and log files. \
+                        Defaults to `./test`. If setup was run with a custom directory, \
+                        the same directory must be provided to cleanup, otherwise only \
+                        peer processes will be destroyed")
+    parser.add_argument("--host_addr", "-a", default="127.0.0.1", type=ipaddress.IPv4Address,
+                        help="IP address of the host machine. Used in trusted peer \
+                        generation. Defaults to localhost without port specified")
```

Comment:
> without port specified

I don't quite get what it means..  AFAIK `ipaddress.IPv4Address` doesn't allow the port to be specified

Reply:
What I meant here is just that the port should not supplied in the host string after the address itself. I'll rephrase it to make it more clear.